cms/static/cms/js/modules/cms.message.js

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright https://github.com/divio/django-cms
+ */
+
+// #############################################################################
+// NAMESPACES
+/**
+ * @module CMS
+ */
+var CMS = window.CMS || {};
+
+// #############################################################################
+// MODAL
+(function ($) {
+    'use strict';
+
+    // shorthand for jQuery(document).ready();
+    $(function () {
+        /**
+         * Displays a message underneath the toolbar
+         *
+         * @class Message
+         * @namespace CMS
+         */
+        CMS.Message = new CMS.Class({
+
+            options: {
+                messageDuration: 300,
+                messageDelay: 3000
+            },
+
+            initialize: function initialize(options) {
+                this.options = $.extend(true, {}, this.options, options);
+                this.config = CMS.config;
+                this.settings = CMS.settings;
+
+                // elements
+                this._setupUI();
+            },
+
+            /**
+             * Stores all jQuery references within `this.ui`.
+             *
+             * @method _setupUI
+             * @private
+             */
+            _setupUI: function _setupUI() {
+                var container = $('#cms-toolbar');
+                this.ui = {
+                    container: container,
+                    body: $('html'),
+                    toolbar: container.find('.cms-toolbar'),
+                    messages: container.find('.cms-messages')
+                };
+            },
+
+            /**
+             * Opens a message window underneath the toolbar.
+             *
+             * @method open
+             * @param opts
+             * @param opts.message {String|HTMLNode} message to be displayed
+             * @param [opts.dir='center'] {String} direction to be displayed `center` `left` or `right`
+             * @param [opts.delay=this.options.messageDelay] {Number} delay until message is closed
+             * @param [opts.error] {Boolean} if true sets the style to `.cms-messages-error`
+             */
+            open: function open(opts) {
+                if (!(opts && opts.message)) {
```

Comment:
this part is copied from https://github.com/divio/django-cms/pull/4526/files#diff-f33b2ef979aa78d0eeb29b02871daa5eL457 and changes there are no part within this PR